ca/ecdsa_allow_list.go

```diff
@@ -1,62 +1,29 @@
 package ca
 
 import (
-	"sync"
+	"os"
 
-	"github.com/letsencrypt/boulder/log"
-	"github.com/letsencrypt/boulder/reloader"
 	"github.com/letsencrypt/boulder/strictyaml"
-	"github.com/prometheus/client_golang/prometheus"
 )
 
-// ECDSAAllowList acts as a container for a map of Registration IDs, a
-// mutex, and a file reloader. This allows the map of IDs to be updated
-// safely if changes to the allow list are detected.
+// ECDSAAllowList acts as a container for a map of Registration IDs.
 type ECDSAAllowList struct {
-	sync.RWMutex
 	regIDsMap map[int64]bool
-	reloader  *reloader.Reloader
-	logger    log.Logger
-}
-
-// Update is an exported method (typically specified as a callback to a
-// file reloader) that replaces the inner `regIDsMap` with the contents
-// of a YAML list (as bytes).
-func (e *ECDSAAllowList) Update(contents []byte) error {
-	var regIDs []int64
-	err := strictyaml.Unmarshal(contents, &regIDs)
-	if err != nil {
-		return err
-	}
-	e.Lock()
-	defer e.Unlock()
-	e.regIDsMap = makeRegIDsMap(regIDs)
-	return nil
 }
 
 // permitted checks if ECDSA issuance is permitted for the specified
 // Registration ID.
 func (e *ECDSAAllowList) permitted(regID int64) bool {
-	e.RLock()
-	defer e.RUnlock()
 	return e.regIDsMap[regID]
 }
 
-// length returns the number of entries currently in the allow list.
+// length returns the number of entries currently in the allow list.s
 func (e *ECDSAAllowList) length() int {
-	e.RLock()
-	defer e.RUnlock()
-	return len(e.regIDsMap)
-}
-
-// Stop stops an active allow list reloader. Typically called during
-// boulder-ca shutdown.
-func (e *ECDSAAllowList) Stop() {
-	e.Lock()
-	defer e.Unlock()
-	if e.reloader != nil {
-		e.reloader.Stop()
+	if e == nil || e.regIDsMap == nil {
+		return 0
 	}
+
+	return len(e.regIDsMap)
 }
```

Comment:
If you take my suggestion above, I believe you can do away with the `length` method altogether. It was useful when we were recording stats regarding the list on each (live) reload.

Reply:
Good idea. I've come up with a slightly different solution than the one you proposed.